test/e2e/e2e_test.go

```diff
@@ -767,6 +857,39 @@ func TestE2E(t *testing.T) {
 		}
 	})
 
+	// make sure that deleting service-cert-secret regenerates a working secret again
+	t.Run("serving-cert-secret-delete-data", func(t *testing.T) {
+		serviceName := "metrics"
+		namespace := "openshift-service-ca-operator"
+		ns, err := createTestNamespace(adminClient, "test-"+randSeq(5))
+		defer deleteNamespace(adminClient, ns.Name)
+		if err != nil {
+			t.Fatalf("could not create test namespace: %v", err)
+		}
+		service, err := adminClient.CoreV1().Services(namespace).Get(serviceName, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("fetching service from apiserver failed: %v", err)
+		}
+		secretName, ok := service.ObjectMeta.Annotations[api.ServingCertSecretAnnotation]
+		if !ok {
+			t.Fatalf("secret name not found in service annotations")
+		}
+		oldCertPEM, oldKeyPEM, err := pollForUpdatedServingCert(t, adminClient, namespace, secretName, rotationPollTimeout, nil, nil)
+		if err != nil {
+			t.Fatalf("error retrieving service cert: %v", err)
+		}
+		err = adminClient.CoreV1().Secrets(namespace).Delete(secretName, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("deleting secret %s in namespace %s failed: %v", secretName, namespace, err)
+		}
+		updatedBytes, _, err := pollForUpdatedServingCert(t, adminClient, namespace, secretName, rotationPollTimeout, oldCertPEM, oldKeyPEM)
```

Comment:
Since the secret has been deleted, it's enough to supply `nil` for `oldCertValue` and `oldKeyValue` (obviating the need for the previous call to `pollForUpdatedServingCert` on line 877.

Reply:
i was confused and added this. i understand that it is not needed and removed it.